File: dictionary/src/main/java/org/ru/dictionary/service/LevelService.java
package org.ru.dictionary.service;

import org.ru.dictionary.dto.level.LevelRequestDTO;
import org.ru.dictionary.dto.level.LevelResponseDTO;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.List;

public interface LevelService {
    LevelResponseDTO createLevel(LevelRequestDTO dto, UserDetails userDetails);
    LevelResponseDTO getLevelById(Long levelId);
    List<LevelResponseDTO> getAllLevelsByCourse(Long courseId);
    LevelResponseDTO updateLevel(Long levelId, LevelRequestDTO dto, UserDetails userDetails);
    void deleteLevel(Long levelId, UserDetails userDetails);
}

File: frontend/src/services/api/level.service.ts
import type { LevelRequestDTO, LevelResponseDTO } from '@/shared/types/level'
import { authAxios } from './index'

export const levelService = {
	getAllLevels: async (courseId?: number): Promise<LevelResponseDTO[]> => {
		const response = await authAxios.get<LevelResponseDTO[]>(
			'/api/levels',
			courseId ? { params: { courseId } } : undefined
		)
		return response.data
	},

	getLevelById: async (id: number): Promise<LevelResponseDTO> => {
		const response = await authAxios.get<LevelResponseDTO>(`/api/levels/${id}`)
		return response.data
	},

	createLevel: async (data: LevelRequestDTO): Promise<LevelResponseDTO> => {
		const response = await authAxios.post<LevelResponseDTO>('/api/levels', data)
		return response.data
	},

	updateLevel: async (
		id: number,
		data: LevelRequestDTO
	): Promise<LevelResponseDTO> => {
		const response = await authAxios.put<LevelResponseDTO>(
			`/api/levels/${id}`,
			data
		)
		return response.data
	},

	deleteLevel: async (id: number): Promise<void> => {
		await authAxios.delete(`/api/levels/${id}`)
	},
}
